src/Components/Blazor/WebAssembly.Authentication/src/AuthenticationManager.cs

```diff
@@ -0,0 +1,446 @@
+            {
+                case RemoteAuthenticationActions.Login:
+                    await ProcessLogin(GetReturnUrl(state: null));
+                    break;
+                case RemoteAuthenticationActions.LoginCallback:
+                    await ProcessLoginCallback();
+                    break;
+                case RemoteAuthenticationActions.LoginFailed:
+                    _message = GetErrorMessage();
+                    break;
+                case RemoteAuthenticationActions.Profile:
+                    if (ApplicationPaths.RemoteProfilePath == null)
+                    {
+                        ProfileFragment??= ProfileNotSupportedFragment;
+                    }
+                    else
+                    {
+                        ProfileFragment??= LoginFragment;
+                        await RedirectToProfile();
+                    }
+                    break;
+                case RemoteAuthenticationActions.Register:
+                    if (ApplicationPaths.RemoteRegisterPath == null)
+                    {
+                        RegisterFragment ??= RegisterNotSupportedFragment;
+                    }
+                    else
+                    {
+                        RegisterFragment ??= LoginFragment;
+                    }
+
+                    await RedirectToRegister();
+                    break;
+                case RemoteAuthenticationActions.Logout:
+                    await ProcessLogout(GetReturnUrl(state: null, Navigation.ToAbsoluteUri(ApplicationPaths.LogoutSucceededPath).AbsoluteUri));
+                    break;
+                case RemoteAuthenticationActions.LogoutCallback:
+                    await ProcessLogoutCallback();
+                    break;
+                case RemoteAuthenticationActions.LogoutFailed:
+                    _message = GetErrorMessage();
+                    break;
+                case RemoteAuthenticationActions.LogoutSucceeded:
+                    break;
+                default:
+                    throw new InvalidOperationException($"Invalid action '{Action}'.");
+            }
+        }
+
+        private async Task ProcessLogin(string returnUrl)
+        {
+            AuthenticationState.ReturnUrl = returnUrl;
+            var result = await AuthenticationService.SignInAsync(new RemoteAuthenticationContext<TAuthenticationState>
+            {
+                State = AuthenticationState
+            });
+            switch (result.Status)
+            {
+                case RemoteAuthenticationStatus.Redirect:
+                    break;
+                case RemoteAuthenticationStatus.Success:
+                    Navigation.NavigateTo(returnUrl);
+                    break;
+                case RemoteAuthenticationStatus.Failure:
+                    var uri = Navigation.ToAbsoluteUri($"{ApplicationPaths.LoginFailedPath}?message={result.ErrorMessage}").ToString();
+                    Navigation.NavigateTo(uri);
```

Comment:
Should we really be using `location.replace` for redirections like this? It seems weird that the user could press "back" to replay the completion attempt.

Reply:
Updated to use the new navigation methods